initial_medial_final not including a required dependency

### Expected behavior

To only require to import  `stdx/algorithm.hpp` to be able to use initial_medial_final

``` c++
#include <stdx/algorithm.hpp>

void foo() {
    auto values = std::array {1, 2, 5, 7, 9};
    stdx::initial_medial_final(std::cbegin(values), std::cend(values), init_func, mid_func, final_func);
}
````

### Current behavior

The following error is triggered, solved by including `<stdx/tuple_destructure.hpp>`  in the source file too.

``` bash
error: cannot decompose class type ‘stdx::v1::detail::tuple_impl<std::integer_sequence<long unsigned int, 0, 1>, stdx::v1::detail::index_function_list<>, dgpu_pm2::hello_world::stdx()::<lambda(auto:24)>, const int*>’: its base classes ‘stdx::v1::detail::element<0, dgpu_pm2::hello_world::stdx()::<lambda(auto:24)> >’ and ‘stdx::v1::detail::element<1, const int*>’ have non-static data members
  130 |         auto [op, it] = for_each_butlast(++first, last, mop);
      |              ^~~~~~~~
/home/jloaiza/ws/firmware.dgpu.pm2/build/_deps/cpp-std-extensions-src/include/stdx/algorithm.hpp:134:29: error: could not convert ‘{iop, op, fop}’ from ‘<brace-enclosed initializer list>’ to ‘stdx::v1::for_each_result<dgpu_pm2::hello_world::stdx()::<lambda(auto:23)>, dgpu_pm2::hello_world::stdx()::<lambda(auto:24)>, dgpu_pm2::hello_world::stdx()::<lambda(auto:25)> >’ {aka ‘stdx::v1::tuple<dgpu_pm2::hello_world::stdx()::<lambda(auto:23)>, dgpu_pm2::hello_world::stdx()::<lambda(auto:24)>, dgpu_pm2::hello_world::stdx()::<lambda(auto:25)> >’}
  134 |         return {iop, op, fop};
      |                             ^
      |                             |
      |                             <brace-enclosed initializer list>
```

--- a/include/stdx/algorithm.hpp
+++ b/include/stdx/algorithm.hpp
@@ -127,11 +127,14 @@ CONSTEXPR_INVOKE auto initial_medial_final(FwdIt first, FwdIt last, IOp iop,
     -> for_each_result<IOp, MOp, FOp> {
     if (first != last) {
         iop(*first);
-        auto [op, it] = for_each_butlast(++first, last, mop);
-        if (it != last) {
+        auto r = for_each_butlast(++first, last, mop);
+#if __cplusplus < 202002L
+        using std::get;
+#endif
+        if (auto it = get<1>(r); it != last) {
             fop(*it);
         }
-        return {iop, op, fop};
+        return {iop, get<0>(r), fop};
     }
     return {iop, mop, fop};
 }

